tower-buffer/tests/buffer.rs

```diff
@@ -95,9 +95,9 @@ fn when_inner_fails() {
     ::std::thread::sleep(::std::time::Duration::from_millis(100));
     with_task(|| {
         let e = res1.poll().unwrap_err();
-        if let Error::Closed(e) = e {
+        if let Some(e) = e.downcast_ref::<ClosedError<tower_mock::Error>>() {
             assert!(format!("{:?}", e).contains("poll_ready"));
-            assert_eq!(e.error(), &tower_mock::Error::Other("foobar"));
+            // assert_eq!(e.error(), &tower_mock::Error.downcast_ref::);
```

Comment:
This works but is somewhat ugly:
```suggestion
            assert_eq!(format!("{}", e.error().error()), "foobar".to_owned());
```

Unfortunately we can't downcast `e.error()` to `str`, as `downcast_ref` and `is` only work for `Sized` types, but the impl of `Error` is for `str` and not `&'a str`...

Reply:
That's way better, thanks!